src/containers/Application/ApplicationSummary.tsx

```diff
@@ -10,54 +9,53 @@ import { SectionWrapper } from '../../components/Application'
 import strings from '../../utils/constants'
 import { Button, Header, Message, Container } from 'semantic-ui-react'
 
-interface ApplicationProps {
-  structure: FullStructure
-}
-
-const ApplicationSummary: React.FC<ApplicationProps> = ({ structure }) => {
+const ApplicationSummary: React.FC<ApplicationProps> = ({
+  structure: fullStructure,
+  requestRevalidation,
+}) => {
   const { replace, push } = useRouter()
-  const [shouldRevalidate, setShouldRevalidate] = useState(false)
-  const [isRevalidated, setIsRevalidated] = useState(false)
+  const [error, setError] = useState(false)
+
   const {
     userState: { currentUser },
   } = useUserState()
-  const { error, fullStructure } = useGetFullApplicationStructure({
-    structure,
-    shouldRevalidate,
-  })
 
   const { error: submitError, processing, submitted, submit } = useSubmitApplication({
     serialNumber: fullStructure?.info.serial as string,
     currentUser: currentUser as User,
   })
 
   useEffect(() => {
-    if (!fullStructure || submitted) return
+    if (!fullStructure) return
     // Re-direct if application is not valid
     if (fullStructure.info.firstStrictInvalidPage) {
       const { sectionCode, pageNumber } = fullStructure.info.firstStrictInvalidPage
       replace(`/applicationNEW/${fullStructure.info.serial}/${sectionCode}/Page${pageNumber}`)
     }
-    if (shouldRevalidate) setIsRevalidated(true)
   }, [fullStructure])
 
   const handleSubmit = () => {
-    setShouldRevalidate(true)
+    requestRevalidation &&
+      requestRevalidation(
+        async ({ firstStrictInvalidPage, setStrictSectionPage }: MethodToCallProps) => {
+          if (firstStrictInvalidPage) {
+            const { sectionCode, pageNumber } = firstStrictInvalidPage
+            setStrictSectionPage(firstStrictInvalidPage)
+            replace(`/applicationNEW/${fullStructure.info.serial}/${sectionCode}/Page${pageNumber}`)
+          } else {
+            try {
+              fullStructure?.responsesByCode &&
```

Comment:
hmm any reason to check responsesByCode ?

Reply:
Cos Typescript complains that responsesByCode could be undefined in the next statement.

Reply:
sweet